src/slice.rs

```diff
@@ -103,55 +112,66 @@ pub enum SliceOrIndex {
     },
     /// A single index.
     Index(isize),
+    /// A new axis of length 1.
+    NewAxis,
 }
 
-copy_and_clone!{SliceOrIndex}
+copy_and_clone!{AxisSliceInfo}
 
-impl SliceOrIndex {
+impl AxisSliceInfo {
     /// Returns `true` if `self` is a `Slice` value.
     pub fn is_slice(&self) -> bool {
         match self {
-            &SliceOrIndex::Slice { .. } => true,
+            &AxisSliceInfo::Slice { .. } => true,
             _ => false,
         }
     }
 
     /// Returns `true` if `self` is an `Index` value.
     pub fn is_index(&self) -> bool {
         match self {
-            &SliceOrIndex::Index(_) => true,
+            &AxisSliceInfo::Index(_) => true,
+            _ => false,
+        }
+    }
+
+    /// Returns `true` if `self` is a `NewAxis` value.
+    pub fn is_new_axis(&self) -> bool {
+        match self {
+            &AxisSliceInfo::NewAxis => true,
             _ => false,
         }
     }
 
-    /// Returns a new `SliceOrIndex` with the given step size (multiplied with
+    /// Returns a new `AxisSliceInfo` with the given step size (multiplied with
     /// the previous step size).
     ///
     /// `step` must be nonzero.
     /// (This method checks with a debug assertion that `step` is not zero.)
     #[inline]
     pub fn step_by(self, step: isize) -> Self {
-        debug_assert_ne!(step, 0, "SliceOrIndex::step_by: step must be nonzero");
+        debug_assert_ne!(step, 0, "AxisSliceInfo::step_by: step must be nonzero");
```

Comment:
I guess we want a debug assert here, that we have a slice variant

Reply:
You mean panic in debug mode if `self` is not the `AxisSliceInfo::Slice` variant? That's probably a good idea. We might as well panic in release mode, too, because we have to perform the check regardless so there's no extra performance cost.